src/overlays/actors/ovl_En_Dekubaba/z_en_dekubaba.c

```diff
@@ -121,85 +133,1191 @@ static ColliderJntSphInit D_809E8FC8 = {
+    gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_en_dekubaba.c", 2710),
+              G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gSPDisplayList(POLY_XLU_DISP++, gCircleShadowDL);
+    CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_en_dekubaba.c", 2715);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Dekubaba/func_809E7BB0.s")
+void EnDekubaba_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, void* thisx) {
+    EnDekubaba* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Dekubaba/func_809E7F14.s")
+    if (limbIndex == 1) {
+        Collider_UpdateSpheres(limbIndex, &this->collider);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Dekubaba/func_809E80D8.s")
+void EnDekubaba_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    EnDekubaba* this = THIS;
+    f32 scale;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Dekubaba/func_809E8140.s")
+    if (1) {}
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_en_dekubaba.c", 2752);
+    func_80093D18(globalCtx->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Dekubaba/EnDekubaba_Update.s")
+    if (this->actionFunc != EnDekubaba_DeadStickDrop) {
+        SkelAnime_DrawOpa(globalCtx, this->skelAnime.skeleton, this->skelAnime.jointTable, NULL,
+                          EnDekubaba_PostLimbDraw, this);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Dekubaba/func_809E858C.s")
+        if (this->actionFunc == EnDekubaba_Wait) {
+            EnDekubaba_DrawStemRetracted(this, globalCtx);
+        } else {
+            EnDekubaba_DrawStemExtended(this, globalCtx);
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Dekubaba/func_809E86B8.s")
+        scale = this->size * 0.01f;
+        Matrix_Translate(this->actor.home.pos.x, this->actor.home.pos.y, this->actor.home.pos.z, MTXMODE_NEW);
+        Matrix_RotateY(this->actor.home.rot.y * (M_PI / 0x8000), MTXMODE_APPLY);
+        Matrix_Scale(scale, scale, scale, MTXMODE_APPLY);
+        gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_en_dekubaba.c", 2780),
+                  G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gSPDisplayList(POLY_OPA_DISP++, gDekuBabaBaseLeavesDL);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Dekubaba/func_809E89E4.s")
+        if (this->actionFunc == EnDekubaba_PrunedSomersault) {
+            EnDekubaba_DrawStemBasePruned(this, globalCtx);
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Dekubaba/func_809E8AD8.s")
+        if (this->boundFloor != NULL) {
+            EnDekubaba_DrawBaseShadow(this, globalCtx);
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Dekubaba/func_809E8C0C.s")
+        // Display solid until 40 frames left, then blink until killed.
+    } else if ((this->timer > 40) || (this->timer & 1)) {
```

Comment:
```suggestion
    } else if ((this->timer > 40) || (this->timer % 2 != 0)) {
```
time for modulus rng

Reply:
mod 2 is usually okay as either (and is here), although I don't think there's much to choose between them.

Reply:
I think it makes a bit more sense to use mod 2 here because this is a timer, so it should be clear this is asking for an odd number imo

Reply:
Usually I like modulus better because it's clearer that the intent is to display every two (`% 2`) frames, but it's whatever you like

Reply:
In general we try to use mod when possible, but depending on the type you may be forced to use &.
This is pretty all over the place in the codebase though and will prob be cleaned up in its own pass